pkg/pillar/cmd/zedagent/handlecontent.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedagent
+
+// content info specific parser/utility routines
+
+import (
+	"bytes"
+	"crypto/sha256"
+
+	zconfig "github.com/lf-edge/eve/api/go/config"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+var contentInfoHash []byte
+
+// cipher context parsing routine
+func parseContentInfoConfig(ctx *getconfigContext,
+	config *zconfig.EdgeDevConfig) {
+
+	log.Infof("Started parsing content info config")
+	cfgContentTreeList := config.GetContentInfo()
+	h := sha256.New()
+	for _, cfgContentTree := range cfgContentTreeList {
+		computeConfigElementSha(h, cfgContentTree)
+	}
+	newHash := h.Sum(nil)
+	if bytes.Equal(newHash, contentInfoHash) {
+		return
+	}
+	log.Infof("parseContentInfo: Applying updated config "+
+		"Last Sha: % x, "+
+		"New  Sha: % x, "+
+		"Num of cfgContentInfo: %d",
+		contentInfoHash, newHash, len(cfgContentTreeList))
+
+	contentInfoHash = newHash
+
+	// First look for deleted ones
+	items := ctx.pubContentTreeConfig.GetAll()
+	for idStr := range items {
+		found := false
+		for _, cfgContentTree := range cfgContentTreeList {
+			if cfgContentTree.GetUuid() == idStr {
+				found = true
+				break
+			}
+		}
+		// content tree not found, delete
+		if !found {
+			log.Infof("parseContentInfo: deleting %s\n", idStr)
+			unpublishContentTreeConfig(ctx, idStr)
+		}
+	}
+
+	for _, cfgContentTree := range cfgContentTreeList {
+		contentConfig := new(types.ContentTreeConfig)
+		contentConfig.ContentID, _ = uuid.FromString(cfgContentTree.GetUuid())
+		contentConfig.DatastoreID, _ = uuid.FromString(cfgContentTree.GetDsId())
+		contentConfig.RelativeURL = cfgContentTree.GetURL()
+		contentConfig.Format = cfgContentTree.GetIformat()
+		if cfgContentTree.GetIformat() == zconfig.Format_CONTAINER {
+			contentConfig.IsContainer = true
```

Comment:
I wonder whether we can avoid the IsContainer boolean by insteafd checking Format == CONTAINER where we need to check.

Reply:
Or make IsContainer() a member function for ctconfig and ctstatus?

Reply:
Added IsContainer() function for ctStatus.